src/numerics.jl

```diff
@@ -103,7 +103,7 @@ function compute_evolution!(discretization, co2_concentration_at_step, year_star
             old_time_step = (time_step == 1) ? num_steps_year : time_step - 1
             update_rhs!(rhs, mesh, num_steps_year, time_step, view(annual_temperature, :, old_time_step), model, last_rhs)
                         
-            annual_temperature[:, time_step] = lu_decomposition \ rhs
+            ldiv!(view(annual_temperature, :, time_step), lu_decomposition, rhs)
```

Comment:
```suggestion
            # Use in-place operation `ldiv!` instead of `\` to avoid allocations
            ldiv!(view(annual_temperature, :, time_step), lu_decomposition, rhs)
```